library/std/src/sys/unix/process/process_unix.rs

```diff
@@ -533,3 +610,45 @@ impl fmt::Display for ExitStatus {
         }
     }
 }
+
+#[unstable(feature = "linux_pidfd", issue = "none")]
+#[derive(Debug)]
+pub struct PidFd {
+    fd: libc::c_int,
+}
+
+#[unstable(feature = "linux_pidfd", issue = "none")]
+impl crate::os::unix::io::AsRawFd for PidFd {
+    fn as_raw_fd(&self) -> crate::os::unix::io::RawFd {
+        self.fd
+    }
+}
+
+#[unstable(feature = "linux_pidfd", issue = "none")]
+impl crate::os::unix::io::IntoRawFd for PidFd {
+    fn into_raw_fd(self) -> crate::os::unix::io::RawFd {
+        self.fd
```

Comment:
This ist Missing a Mem::forget(self)

Reply:
Thanks! I fixed it in the most recent commit.